affinity/src/main/java/net/openhft/affinity/impl/VersionHelper.java

```diff
@@ -70,9 +70,12 @@ public boolean majorMinorEquals(final VersionHelper ver) {
 
     public boolean isSameOrNewer(final VersionHelper ver) {
         return ver != null
-                && this.major >= ver.major
-                && this.minor >= ver.minor
-                && this.release >= ver.release;
+                && (this.major > ver.major
+                    || (this.major == ver.major
+                        && this.minor > ver.minor)
+                    || (this.major == ver.major
+                        && this.minor == ver.minor
+                        && this.release >= ver.release));
```

Comment:
could do even simpler, in same style:
```java
&& (this.major > ver.major
    || this.major == ver.major
        && (this.minor > ver.minor
            || this.minor == ver.minor
                && this.release >= ver.release))
```

Reply:
figured it could shorter. can update to this if you think it's still readable